src/app/header/header.component.css

```diff
@@ -1,3 +1,6 @@
+*{
+    transition: background-color 0.5s ease;
+}
```

Comment:
Can you please check if there's a better way of doing the smooth dark mode transition? I'll also research this a bit.